test/core/common/result_test.cpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+#include <spdlog/spdlog.h>
+
+#include "common/result.hpp"
+
+using kagome::expected::Result;
+using kagome::expected::Value;
+using kagome::expected::Error;
+using kagome::expected::NoValueException;
+using kagome::expected::UnwrapException;
+
+/**
+ * Created to detect unnecessary copying of values in Result
+ */
+class T {
+public:
+  T() { spdlog::info("T()"); }
```

Comment:
was it done just for debugging?
Maybe we need stronger tests checking that there is no copying. For example we can run `FAIL(); if copy has happened